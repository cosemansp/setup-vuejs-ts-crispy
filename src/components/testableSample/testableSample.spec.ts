import { mount, mountConfig } from '../../../test/helper';

import TestableSample from './testableSample';

describe('Hello Component', () => {
  it('should have a default "title"', () => {
    const component = new TestableSample();
    expect(component.title).toMatch('I am testable');
  });

  it('should renders correctly', () => {
    // render component with child components
    const fixture = mount(TestableSample);
    fixture.setData({ title: 'I am testable' });
    expect(fixture).toMatchSnapshot();
  });

  it('should renders config', () => {
    const componentConfig = {
      template: `
        <div>
          <testableSample ref="component"></testableSample>
        </div>
      `,
      components: {
        testableSample: TestableSample,
      },
    };
    const fixture = mountConfig(componentConfig);
    expect(fixture).toMatchSnapshot();
    expect(fixture.vm.$refs.component).toBeTruthy();
    expect(fixture.vm.$refs.component.title).toMatch('I am testable');
  });
});
