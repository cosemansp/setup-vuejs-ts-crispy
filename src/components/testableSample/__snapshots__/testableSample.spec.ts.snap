// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello Component should renders config 1`] = `
<div>
  <div>
    <h1>I am testable</h1>
    <div>I am a child</div>
  </div>
</div>
`;

exports[`Hello Component should renders correctly 1`] = `
<div>
  <h1>I am testable</h1>
  <div>I am a child</div>
</div>
`;
